pykern/pkio.py

```diff
@@ -125,6 +125,30 @@ def has_file_extension(filename, to_check):
     return e in to_check
 
 
+def is_binary(filename, threshold=0.30):
+    """Uses heuristics to guess whether file is binary data
+
+    Args:
+        filename (str|py.path.local): file to check
+        threshold (float): allowed percentage of assumed non-textual characters
+
+    Returns:
+        bool: True if file is likely binary data
+    """
+    if threshold >= 1:
```

Comment:
We don't normally test this. However, if you do, it should not be allowed to be negative either.